tests/testthat/test-dup.R

```diff
@@ -13,3 +13,20 @@ test_that("0 duplicates equals empty string", {
 test_that("uses tidyverse recycling rules", {
   expect_error(str_dup(1:2, 1:3), class = "vctrs_error_incompatible_size")
 })
+
+test_that("uses sep argument", {
+  expect_equal(str_dup("a", 3, sep = ";"), "a;a;a")
```

Comment:
What happens if you do `str_dup(character(), 1, sep = ":")`?

Reply:
added as a test. It returns character()